plugins/content/iwcsocialmeta/iwcsocialmeta.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Content.socialcustommetatag
+ * @copyright   Copyright (C) 2016 - Tasolglobal. All rights reserved.
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ * @since       3.6
+ */
+
+defined('_JEXEC') or die;
+use Joomla\Registry\Registry;
+JLoader::import('joomla.application.component.model');
+
+/**
+ * Iwcsocialmeta class for the Social Meta Plugin.
+ *
+ * @since  3.6
+ */
+
+class PlgContentIwcsocialmeta extends JPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 * @since  3.6
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Runs on content preparation
+	 *
+	 * @param   string  $context  The context for the data
+	 * @param   object  $data     An object containing the data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentPrepareData($context, $data)
+	{
+		/*if (!in_array($context, array('com_content.article','com_contact.contact')))*/
+		if (!in_array($context, array('com_content.article')))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
```

Comment:
`$app` is used only once in your function scope. No needs to create a variable if it is used only for one time. Variable should be created only if it is used multiple times or needs to improve readability.

```php
if (JFactory::getApplication()->isAdmin())
```
is enough to write.